br refactor: translate log backup and pitr (#11395)
pingcap/docs: br/br-pitr-manual.md

---

> title: TiDB Log Backup and PITR Command-line Manual

Reviewer: ```suggestion
title: TiDB Log Backup and PITR Command Manual
```

> summary: Learn about the commands of TiDB log backup and Point-in-time recovery.

Reviewer: ```suggestion
summary: Learn about the commands of TiDB log backup and point-in-time recovery.
```

aliases: ['/tidb/dev/br-log-command-line/']
---

> # TiDB Log Backup and PITR Command-line Manual

Reviewer: ```suggestion
# TiDB Log Backup and PITR Command Manual
```

> This document describes commands of TiDB log backup the PITR (Point-in-time recovery).

Reviewer: ```suggestion
This document describes the commands used in TiDB log backup point-in-time recovery (PITR).
```

For more information about log backup and PITR, you can refer to the following documents:

- [TiDB Log Backup and PITR Guide](/br/br-pitr-guide.md)
- [TiDB Back up and Restore Use Cases](/br/backup-and-restore-use-cases.md)

## Perform log backup

> You can perform log backup and manage backup tasks using the `br log` command.

Reviewer: ```suggestion
You can start and manage log backup using the `br log` command.
```

### Start a backup task

You can run the `br log start` command to start a log backup task. This task runs in the background of your TiDB cluster and automatically backs up the change log of KV storage to the backup storage.

Run `br log start --help` to see the help information:

```shell
./br log start --help
start a log backup task

Usage:
  br log start [flags]

Flags:
  -h, --help               help for start
  --start-ts string        usually equals last full backupTS, used for backup log. Default value is current ts. support TSO or datetime, e.g. '400036290571534337' or '2018-05-11 01:42:23+0800'.
  --task-name string       The task name for the backup log task.

Global Flags:
 --ca string                  CA certificate path for TLS connection
 --cert string                Certificate path for TLS connection
 --key string                 Private key path for TLS connection
 -u, --pd strings             PD address (default [127.0.0.1:2379])
 -s, --storage string         specify the url where backup storage, eg, "s3://bucket/path/prefix"

```

The example output only shows the common parameters. These parameters are described as follows:

> - `--start-ts`: specifies the start timestamp for the log backup. If this is not specified, the backup program uses the current time as `start-ts`.

Reviewer: ```suggestion
- `--start-ts`: specifies the start timestamp for the log backup. If this parameter is not specified, the backup program uses the current time as `start-ts`.
```

Full Restore <--------------------------------------------------------------------------------------------------------------------------------------------------------> 100.00%
*** ***["Full Restore success summary"] ****** [total-take=3.112928252s] [restore-data-size(after-compressed)=5.056kB] [Size=5056] [BackupTS=434693927394607136] [total-kv=4] [total-kv-size=290B] [average-speed=93.16B/s]
Restore Meta Files <--------------------------------------------------------------------------------------------------------------------------------------------------> 100.00%
Restore KV Files <----------------------------------------------------------------------------------------------------------------------------------------------------> 100.00%
"restore log success summary"] [total-take=192.955533ms] [restore-from=434693681289625602] [restore-to=434693753549881345] [total-kv-count=33] [total-size=21551]
```

> **Note:**
>
> - You cannot restore the log backup data of a certain time period repeatedly. If you restore the log backup data of a range `[t1=10, t2=20)` repeatedly, the restored data might be inconsistent.

> > - When you restore log data of different time periods in multiple batches, you must ensure the log data is restored in consecutive order. If you restore the log backup data of `[t1, t2)`, `[t2, t3)`, and `[t3, t4)` in consecutive order, the restored data is consistent. However, if you restore `[t1, t2)` and then skip `[t2, t3)` to restore `[t3, t4)`, the restored data might be inconsistent.

Reviewer: ```suggestion
> - When you restore log data of different time periods in multiple batches, ensure that the log data is restored in consecutive order. If you restore the log backup data of `[t1, t2)`, `[t2, t3)`, and `[t3, t4)` in consecutive order, the restored data is consistent. However, if you restore `[t1, t2)` and then skip `[t2, t3)` to restore `[t3, t4)`, the restored data might be inconsistent.
```
